client/core/trade.go

```diff
@@ -32,13 +32,25 @@ func (err ExpirationErr) Error() string { return string(err) }
 // A matchTracker is used to negotiate a match.
 type matchTracker struct {
 	db.MetaMatch
-	failErr        error
-	refundErr      error
-	prefix         *order.Prefix
-	trade          *order.Trade
-	counterSwap    asset.AuditInfo
-	contractExpiry time.Time // for one time logging in isRefundable, may never be set
-	id             order.MatchID
+	id          order.MatchID
+	failErr     error
+	refundErr   error
+	prefix      *order.Prefix
+	trade       *order.Trade
+	counterSwap asset.AuditInfo
+
+	// The following fields facilitate useful logging, while not being spammy.
+
+	// counterConfirms records the last known confirms of the counterparty swap.
+	// This is set in isSwappable for taker, isRedeemable for maker. -1 means
+	// the confirms have not yet been checked (or logged).
+	counterConfirms int64
+	// lastExpireDur is the most recently logged time until expiry of the
+	// party's own contract. This may be negative if expiry has passed, but it
+	// is not yet refundable due to other consensus rules. This is set in
+	// isRefundable. Initialize this to a very large value to guarantee that it
+	// will be logged on the first check or when 0.
+	lastExireDur time.Duration
```

Comment:
```suggestion
	lastExpireDur time.Duration
```

Reply:
I think we could look at how we trigger these checks, but a match might become swapable after some counterparty action rather than something mined.  e.g. extracting the secret from the maker's bcast redeem, so you can bcast your own right away rather than waiting for a block

Reply:
Yeah, the match would be redeemable in that case (not swappable) and I think that's the only match step (aside the maker's init swap) that does not wait for some confirmations.

Is it something you want to pursue in this PR, or should I look into it separately?

Reply:
Please take that up if you like.  I need to pursue other items presently, including your auto-redeem PR